File: assign2/ocaml/src/recursive.ml
let rec recursiveFib position =
  match position with
  | 0 -> [1]
  | 1 -> [1; 1]
  | _ ->
      let series = recursiveFib (position - 1) in
      let length = List.length series in
      let cur = List.nth series (length - 1) in
      let prev = List.nth series (length - 2) in
      series @ [prev + cur]

File: assign2/erlang/src/recursive.erl
-module(recursive).
-export([recursiveFib/1]).

recursiveFib(0) -> [1];
recursiveFib(1) -> [1, 1];
recursiveFib(Position) when Position >= 2 -> 
  Prev = recursiveFib(Position - 1),
  Prev ++ [lists:nth(Position, Prev) + lists:nth(Position - 1, Prev)].
